cmd/diagnostic_test.go

```diff
@@ -171,26 +198,114 @@ token: mytoken
 				Eventually(session).Should(gexec.Exit(0))
 			})
 		})
+	})
 
-		Describe("whoami returns a user", func() {
-			BeforeEach(func() {
-				_, err := config.Write([]byte(`token: mytoken`))
-				Expect(err).ToNot(HaveOccurred())
-				Expect(config.Close()).To(Succeed())
-			})
+	Describe("whoami returns a user", func() {
+		var (
+			tempHome        string
+			command         *exec.Cmd
+			testServer      *ghttp.Server
+			defaultEndpoint = "graphql-unstable"
+			config          *os.File
+		)
 
-			It("print success", func() {
-				session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
-				Expect(err).ShouldNot(HaveOccurred())
-				Eventually(session.Err.Contents()).Should(BeEmpty())
-				Eventually(session.Out).Should(gbytes.Say(
-					fmt.Sprintf("API host: %s", defaultHost)))
-				Eventually(session.Out).Should(gbytes.Say(
-					fmt.Sprintf("API endpoint: %s", testServer.URL())))
-				Eventually(session.Out).Should(gbytes.Say("OK, got a token."))
-				Eventually(session.Out).Should(gbytes.Say("Hello, zzak."))
-				Eventually(session).Should(gexec.Exit(0))
-			})
+		BeforeEach(func() {
+			var err error
```

Comment:
Just a general comment, not specific to this PR: we might want this behavior for all of the CLI integration tests. I remember hitting a development issue with my CLI config file interfering with the behavior of the test suite, causing it to hit production.

Reply:
(Clarification/Follow-up from Slack): It looks like this test setup is creating a temporary config file and configuring the CLI via `command.Env` to use that instead of the default config path. I think we may want this for all tests that make a GraphQL request.

I *think* the weird behavior I saw (months ago) was in tests without `command.Env`, a setting in my `~/.circleci/cli.yml` was overriding the CLI flag we use to set the mock server endpoint. It was either `hosts` or `endpoint` having a URL in my config. I think that usage is now deprecated, but this could be an issue for current CircleCI users who might want to contribute to the CLI.

Reply:
Thank you for the clarification.

For tests that write to config, we use an isolated environment to ensure the user settings aren't affected. We use this pattern in a few other places that I thought were causing similar issues to the one you reported -- but am unaware if this is still a problem.

I agree that probably all tests should run in a clean environment, but that is probably a separate conversation we could have outside this PR.